tests/eas/acceptance.py

```diff
@@ -252,130 +252,39 @@ class OffloadMigrationAndIdlePull(unittest.TestCase):
 
     """
 
-    @classmethod
-    def setUpClass(cls):
-        cls.params = {}
-        cls.env = TestEnv(test_conf=TEST_CONF)
-        cls.trace_file = os.path.join(cls.env.res_dir, "offload_idle_pull.dat")
-        cls.log_file = os.path.join(cls.env.res_dir, "offload_idle_pull.json")
-        cls.early_starters = []
-        cls.migrators = []
-        cls.num_tasks = len(cls.env.target.bl.bigs)
-        cls.populate_tasks()
-        local_setup(cls.env)
-        cls.run_workload()
-
-        cls.trace = trappy.FTrace(cls.trace_file)
-        cls.m_assert = SchedMultiAssert(cls.trace, cls.env.topology,
-                                        execnames=cls.migrators)
-        cls.e_assert = SchedMultiAssert(cls.trace, cls.env.topology,
-                                        execnames=cls.early_starters)
-
-        all_tasks = cls.early_starters + cls.migrators
-        cls.a_assert = SchedMultiAssert(cls.trace, cls.env.topology,
-                                        execnames=all_tasks)
-        cls.offset = cls.get_offset()
-
-        cls.end_times = cls.calculate_end_times()
-        cls.log_fh = open(os.path.join(cls.env.res_dir, cls.log_file), "w")
-
-    @classmethod
-    def tearDownClass(cls):
-        cls.log_fh.close()
-
-    @classmethod
-    def populate_tasks(cls):
-        migrator_workload = BIG_WORKLOAD.copy()
-        migrator_workload["duration_s"] = 9
-        migrator_workload["delay_s"] = OFFLOAD_MIGRATION_MIGRATOR_DELAY
-
-        for idx in range(cls.num_tasks):
-            task = "early_starters" + str(idx)
-            cls.params[task] = Periodic(**BIG_WORKLOAD).get()
-            cls.early_starters.append(task)
-
-            # Tasks that will be idle pulled
-            task = "migrator" + str(idx)
-            cls.params[task] = Periodic(**migrator_workload).get()
-            cls.migrators.append(task)
+    conf_basename = "offload_idle_pull.config"
 
-    @classmethod
-    def run_workload(cls):
-
-        wload = RTA(
-            cls.env.target,
-            "offload_idle_pull",
-            calibration=cls.env.calibration())
-        wload.conf(kind="profile", params=cls.params)
-        cls.env.ftrace.start()
-        wload.run(
-            out_dir=cls.env.res_dir,
-            background=False)
-        cls.env.ftrace.stop()
-        trace = cls.env.ftrace.get_trace(cls.trace_file)
-
-    @classmethod
-    def get_offset(cls):
-        task_start_times = cls.a_assert.getStartTime().values()
-        return min([t['starttime'] for t in task_start_times])
+    @experiment_test
+    def test_first_cpu(self, experiment, tasks):
+        """Offload Migration and Idle Pull: Test First CPU"""
+        self._do_test_first_cpu(experiment, tasks)
 
     @classmethod
-    def calculate_end_times(cls):
+    def calculate_end_times(cls, experiment):
```

Comment:
I think in general it would be lovely to have derived classes which exposes only "test_" methods.